slycot/analysis.py

```diff
@@ -475,6 +475,108 @@ def ab08nd(n,m,p,A,B,C,D,equil='N',tol=0,ldwork=None):
         raise e
     return out[:-1]
 
+def ab08nz(n, m, p, A, B, C, D, equil='N', tol=0., lzwork=None):
+    """ nu,rank,dinfz,nkror,nkrol,infz,kronr,kronl,Af,Bf = ab08nz(n,m,p,A,B,C,D,[equil,tol,lzwork])
+
+    To construct for a linear multivariable system described by a state-space
+    model (A,B,C,D) a regular pencil (Af - lambda*Bf ) which has the invariant
+    zeros of the system as generalized eigenvalues.
+    The routine also computes the orders of the infinite zeros and the
+    right and left Kronecker indices of the system (A,B,C,D).
+
+    Required arguments:
+        n : input int
+            The number of state variables.  n >= 0.
+        m : input int
+            The number of system inputs.  m >= 0.
+        p : input int
+            The number of system outputs.  p >= 0.
+        A : input rank-2 array('d') with bounds (n,n)
+            The leading n-by-n part of this array must contain the state
+            dynamics matrix A of the system.
+        B : input rank-2 array('d') with bounds (n,m)
+            The leading n-by-m part of this array must contain the input/state
+            matrix B of the system.
+        C : input rank-2 array('d') with bounds (p,n)
+            The leading p-by-n part of this array must contain the state/output
+            matrix C of the system.
+        D : input rank-2 array('d') with bounds (p,m)
+            The leading p-by-m part of this array must contain the direct
+            transmission matrix D of the system.
+    Optional arguments:
+        equil := 'N' input string(len=1)
+            Specifies whether the user wishes to balance the compound matrix
+            as follows:
+            = 'S':  Perform balancing (scaling);
+            = 'N':  Do not perform balancing.
+        tol := 0.0 input float
+            A tolerance used in rank decisions to determine the effective rank,
+            which is defined as the order of the largest leading (or trailing)
+            triangular submatrix in the QR (or RQ) factorization with column
+            (or row) pivoting whose estimated condition number is less than 1/tol.
```

Comment:
Maybe add something about the lower bound on tol, say "if tol is too small (including the default value of 0), a tolerance based on machine precision and problem size is used" ? Otherwise the combination of default `tol = 0` and `1/tol` is a bit confusing, and I think misleading.

Reply:
I re-added the paragraph from the Fortran comment in https://github.com/bnavigator/Slycot/tree/pr96-lytex-complex